### DISCUSSION TITLE
Can gmx_MMPBSA output data at preset intervals of a certain number of frames?

### DISCUSSION THREAD

**Discussion by alanjchang** [Valdes-Tresanco-MS/gmx_MMPBSA#527] [Q&A]
My goal is to determine how many frames I need to run gmx_MMPBSA for so that my free energy values for a protein-aptamer complex converge. I have run my GROMACS MD simulation for 10ns (corresponds to ~1000 frames). To achieve my goal, for a protein-aptamer complex, I am currently running gmx_MMPBSA for 100, 200, 300, ..., 1000 frames separately to see if the free energy values converge. This, however, is very computationally expensive, especially given that some of my complexes are large. Is there a way for gmx_MMPBSA to output free energy data at intervals of one gmx_MMPBSA run. For example, I would just run gmx_MMPBSA once for 1000 frames, but every 100 frames, gmx_MMPBSA would output the free energy values it has calculated up until that point. This would be computationally cheaper.

**Comment by marioernestovaldes** [comment]
Every command line in the tutorial includes a way to generate a csv file with all the energy values. I assume you can run 1000 frames and then use python or any other programming language to process all the csv files in batch

https://valdes-tresanco-ms.github.io/gmx_MMPBSA/dev/examples/Protein_DNA/

**Reply by alanjchang** [reply]
I'm not sure if this answers my original question. What can I put in the command line or how can I change the input file so that in ONE gmx_MMPBSA run for 1000 frames, an output file is generated at every 100 frame interval (so at 100, 200, 300, ..., 1000 frames of the SINGLE gmx_MMPBSA run)?

**Reply by Valdes-Tresanco-MS** [reply]
You must only add the `-eo` and/or `-deo` flags to the command line. This will generate one or two CSV files with the energy for terms per frame. Since the calculations are deterministic these files will give you all the information you need. For example, if you want the first 100 frames, extract the first 100 frames, if you wish the odd frames, then select frames 1, 3, 5... and so on. For this, probably you need to know Python, R, Matlab, or Excel. gmx_MMPBSA doesn't manage this kind of analysis.

**Reply by alanjchang** [reply]
In other words, gmx_MMPBSA can only generate one output CSV file (or .dat file). If I request the output data for the first 1000 frames, gmx_MMPBSA cannot simultaneously output results for the first 100, 200, 300, ... < 1000 frames. I would have to execute another gmx_MMPBSA run to get the output data for those numbers of frames.

**Reply by marioernestovaldes** [reply]
There is a way to output energy values at those specific frames using the interval variable in the input file

https://valdes-tresanco-ms.github.io/gmx_MMPBSA/dev/input_file/#basic-input-options

**Reply by Valdes-Tresanco-MS** [reply]
It doesn't make sense for the average user to do calculations for multiple intervals at the same time. Your case is particular, so you must adapt the output to make it work for you. From the CSV file, you can do your calculations manually using the tools I mentioned above. Now, if you want to compute all the statistics we present in the output, then you will have to run multiple calculations. From my perspective, I think you could work with the CSV file and the results that appear in TOTAL. You would have just 1000 numerical values ​​and in Excel, you can do the calculations for 100, 200, and 300 frames. Here you have the options, it is up to you to choose the one you consider most convenient.
